ios/FluentUI/Card Nudge/CardNudge.swift

```diff
@@ -40,14 +40,24 @@ public typealias CardNudgeButtonAction = ((_ state: MSFCardNudgeState) -> Void)
 
     /// Action to be dispatched by the dismiss ("close") button on the trailing edge of the control.
     @objc var dismissButtonAction: CardNudgeButtonAction? { get set }
+
+    /// Design token set for this control, to use in place of the control's default Fluent tokens.
+    @objc var overrideTokens: CardNudgeTokens? { get set }
 }
 
 /// View that represents the CardNudge.
-public struct CardNudge: View {
-    @Environment(\.theme) var theme: FluentUIStyle
-    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+public struct CardNudge: TokenizedControl {
+    public static func defaultTokens() -> CardNudgeTokens {
+        return CardNudgeTokens()
+    }
```

Comment:
If the client is supposed to subclass the tokens, what's the point of having this? They're supposed to have their own class, right? Why do we need to provide a way to get the default? Is the idea is they could get a default token object, modify that, then pass that back in to the class? Are we just giving them two ways to have custom tokens?

Reply:
Also, if the token class is open, why not just have them call the init for it themselves? Seems like we could get away with just overrideTokens in TokenizedControl. As long as we have an example of customizing the default tokens by initializing them and passing them in to the control somewhere in our demo app, we could avoid adding a public static function to all of our controls.

Reply:
Thanks for pointing this out. I was just rethinking this part of the system this morning: I think it's a little more complicated than I'd like, but it's also a complicated problem in general.

There are three places that a component (e.g. `CardNudge`) can get tokens, in priority order:
1. A custom token set from the `overrideTokens` property on its `state`.
2. A custom token set from the current `FluentTheme` object.
3. The default token set provided by Fluent.

This function is part of the mechanism for fetching and cacheing option three inside the FluentTheme - you can see the callback mechanism inside `FluentTheme.tokens(for:)` that invokes this static method if no custom tokens are availble. But I think the set of back-and-forth callbacks between `FluentTheme` and its `TokenizedControl`s is hard to follow. I'll think about simpler alternatives, but I'm also open to suggestions if anything comes to mind!

Reply:
FYI @huwilkes my response to your comment broke certain scenarios; the `FluentTheme` _needs_ to pre-configure the default tokens, to ensure they're referencing the correct `GlobalTokens` and `AliasTokens` instances.

My latest change 3b7f57a is an attempt at a compromise: there's no more public `defaultTokens()` method, but all tokens still pass through the `FluentTheme.tokens(for:)` API. Let me know if this version, along with the comments I added, make sense!

Reply:
Ah, gotcha. That makes sense. The new implementations look good to me, but the comment about the return of `func tokens<T: TokenizedControlInternal>(for control: T) -> T.TokenType` isn't right anymore. And `/// - Parameter control: The control to fetch controls for.` should be `/// - Parameter control: The control to fetch tokens for.`. And a nit: The three different tokens do have different priorities, but I'm not sure they're really in a queue. I think just "Priority for token retrieval" should work fine, though.

Reply:
Great catch! Thanks, updated.